### PR TITLE
Fixed issue with using an ability in wrong direction

### PR DESCRIPTION
**Pull request by Badulundabad**

### CODE REVIEW

File: CSharpSourceCode/Battle/CrosshairMissionBehavior/CustomCrosshairMissionBehavior.cs [hunk 1]
```diff
@@ -140,18 +140,37 @@ private void UpdateAbilityCrosshairVisibility()
...
+                    }
+                    else
+                    {
+                        abilityCrosshair.Hide();
+                    }
                 }
             }
         }
 
+        private bool IsCorrectDirection()
```
Comment (Z3rca) [inline CSharpSourceCode/Battle/CrosshairMissionBehavior/CustomCrosshairMissionBehavior.cs]: the name can be misleading. I would call it IsInFrontOfCharacter)

File: CSharpSourceCode/Battle/CrosshairMissionBehavior/CustomCrosshairMissionBehavior.cs [hunk 2]
```diff
@@ -140,18 +140,37 @@ private void UpdateAbilityCrosshairVisibility()
...
+            var frame1 = Agent.Main.LookFrame;
+            frame1.rotation.f.z = 0;
+            frame1.Advance(1);
+
+            var frame2 = Agent.Main.LookFrame;
+            frame2.rotation.f = Agent.Main.MountAgent.GetMovementDirection();
+            frame2.rotation.f.z = 0;
+            frame2.Advance(1);
+
+            return frame1.origin.Distance(frame2.origin) < 1.2f;
```
Comment (Z3rca) [inline CSharpSourceCode/Battle/CrosshairMissionBehavior/CustomCrosshairMissionBehavior.cs]: Sorry if I might have mistaken something, but the math looks wrong to me. Refractor this to a method called "CalculateAngleBetweenVectors" or similar. inside this you might then make a proper calculation of the vector. like here in  https://www.omnicalculator.com/math/angle-between-two-vectors

**Review by Z3rca — CHANGES_REQUESTED** [review]
Please check out if, the calculation for the angle is correct. what you ultimately search for is an angle difference, not the distance between two vectors.

File: CSharpSourceCode/Battle/CrosshairMissionBehavior/CustomCrosshairMissionBehavior.cs (hunk 1 above)
Comment (Badulundabad) [reply CSharpSourceCode/Battle/CrosshairMissionBehavior/CustomCrosshairMissionBehavior.cs]: Yeah, i wanted to call it the same, but actually player can cast ability to a side, anyway i'll think another name up

File: CSharpSourceCode/Battle/CrosshairMissionBehavior/CustomCrosshairMissionBehavior.cs (hunk 2 above)
Comment (Badulundabad) [reply CSharpSourceCode/Battle/CrosshairMissionBehavior/CustomCrosshairMissionBehavior.cs]: Yep, i created this solution quickly and not so properly if talking about formula and naming. Thanks for your advice, i'll use it.

**Review by Z3rca — APPROVED** [review]

**Review by martkartasev — APPROVED** [review]

**Pull request merged**